oci/layout/oci_transport.go

```diff
@@ -228,7 +247,75 @@ func (ref ociReference) NewImageDestination(ctx context.Context, sys *types.Syst
 
 // DeleteImage deletes the named image from the registry, if supported.
 func (ref ociReference) DeleteImage(ctx context.Context, sys *types.SystemContext) error {
-	return errors.New("Deleting images not implemented for oci: images")
+
+	// Get the manifest for the image
+	manifestDescriptor, err := ref.getManifestDescriptor()
+	if err != nil {
+		return err
+	}
+
+	manifest, err := ref.getManifest(manifestDescriptor)
+	if err != nil {
+		return err
+	}
+
+	// Get all the layers used by all other images
+	index, err := ref.getIndex()
+	if err != nil {
+		return err
+	}
+	layersUsedByOtherImages := make(map[digest.Digest]bool, len(index.Manifests))
```

Comment:
I think this could be `internal/set.Set[digest.Digest]`

Reply:
Didn't see this, will do.